test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -879,6 +879,30 @@ TEST_P(SslSocketTest, GetUriWithUriSan) {
                .setExpectedSerialNumber(TEST_SAN_URI_CERT_SERIAL));
 }
 
+TEST_P(SslSocketTest, IpSan) {
+  const std::string client_ctx_yaml =
+      fmt::format(R"EOF(
+  common_tls_context:
+    validation_context:
+      trusted_ca:
+        filename: "{{{{ test_rundir }}}}/test/config/integration/certs/upstreamcacert.pem"
+      verify_subject_alt_name: {}
+)EOF",
+                  Network::Test::getLoopbackAddressString(GetParam()));
```

Comment:
The SAN is irrelevant to which version of IP the connection is on, so here we lose the coverage for IPv6 SAN, not a big deal though.

Reply:
How do you mean? I agree it's irrelevant but this test will test both cases including the IPv6 SAN?

Reply:
@lizan's point is that this verification isn't platform-dependent (i.e. you can configure `verify_subject_alt_name: "::1"` which will validate IPv6 certificate on a system that doesn't support IPv6). This also means that you don't test that both work on IPv4-only systems.

I'd prefer if you had explicit tests for both: `127.0.0.1` and `::1` instead of using `GetParam()`.

Reply:
done